tests/01-actions.spec.ts

```diff
@@ -737,27 +737,51 @@ describe('BalenaHupActionUtils', () => {
 					before: '6.0.0',
 					cutoff: '6.0.38',
 					takeover: '6.0.39',
+					// pre-unification version for dev mode device when called by balena-proxy
+					takeoverDev: '6.0.39.dev',
+					takeoverProd: '6.0.39.prod',
```

Comment:
Is this really how how the semvers reache hup-action-utils?
I'm asking primarily since these are invalid semvers.
Maybe we could have more realistic ones like:
```suggestion
					takeoverNoRevDev: '6.0.39+dev',
					takeoverNoRevProd: '6.0.39+prod',
					takeoverRev1Dev: '6.0.39+rev1.dev',
					takeoverRev1Prod: '6.0.39+rev1.prod',
					takeoverInvalidSemverDev: '6.0.39.dev',
					takeoverInvalidSemverProd: '6.0.39.prod',
```

Reply:
@thgreasi The semver from balena-proxy is generated [here](https://github.com/balena-io/balena-proxy/blob/f7c2ba3bddbff5b726233c5869642b80db787a85/src/common/version-utils.ts#L54). IOW, the current device version is sent as:
```
`${deviceInfo.osVersion}.${deviceInfo.osVariant}`
```
where these values are `os_version` and `os_variant` from the API device model.

I expect that the `takeoverNoRev*` and `takeoverRev1*` variables will test as not equal to '6.0.39'. Only the takeoverInvalidSemver* variables will test as equal to '6.0.39'.

Reply:
Pushed a commit that includes the gist of the comment.